src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -135,4 +141,21 @@ private function getPrettyName($controller): string
 
         return $controller;
     }
+
+    /**
+     * @param \Throwable[] $valueResolverExceptions
+     */
+    private function constructFailedToResolveException($controllerName, string $argumentName, array $valueResolverExceptions): \RuntimeException
+    {
+        $reasons = [];
+        $reasonCounter = 1;
+        foreach ($valueResolverExceptions as $e) {
+            $reasons[] = "{$reasonCounter}) ".$e->getMessage();
+            ++$reasonCounter;
+        }
+
+        $reasons[] = $reasonCounter.') Either the argument is nullable and no null value has been provided, no default value has been provided or there is a non-optional argument after this one.';
+
+        return new \RuntimeException(sprintf('Controller "%s" requires the "$%s" argument that could not be resolved. Possible reasons: %s', $controllerName, $argumentName, implode(' ', $reasons)));
```

Comment:
let's remove the "Possible reasons: %x)" prefix when there's only one possibility?

Reply:
Makes sense 👍